com.unity.shadergraph/Editor/Drawing/Inspector/SGPropertyDrawer.cs

```diff
@@ -0,0 +1,1224 @@
+            row.Add((Toggle)propertyToggle, (toggle) =>
+            {
+                toggle.value = fieldToDraw;
+            });
+
+            if (valueChangedCallback != null)
+            {
+                var toggle = (Toggle) propertyToggle;
+                toggle.OnToggleChanged(evt => valueChangedCallback(evt.newValue));
+            }
+
+            row.styleSheets.Add(Resources.Load<StyleSheet>("Styles/PropertyRow"));
+            return row;
+        }
+
+        public Action inspectorUpdateDelegate { get; set; }
+
+        public VisualElement DrawProperty(
+            PropertyInfo propertyInfo,
+            object actualObject,
+            InspectableAttribute attribute)
+        {
+            return this.CreateGUI(
+                // Use the setter from the provided property as the callback
+                newBoolValue => propertyInfo.GetSetMethod(true).Invoke(actualObject, new object[] {newBoolValue}),
+                (bool) propertyInfo.GetValue(actualObject),
+                attribute.labelName,
+                out var propertyVisualElement);
+        }
+    }
+
+    [SGPropertyDrawer(typeof(string))]
+    class TextPropertyDrawer : IPropertyDrawer
+    {
+        internal delegate void ValueChangedCallback(string newValue);
+
+        internal VisualElement CreateGUI(
+            ValueChangedCallback valueChangedCallback,
+            string fieldToDraw,
+            string labelName,
+            out VisualElement propertyTextField,
+            int indentLevel = 0)
+        {
+            var propertyRow = new PropertyRow(PropertyDrawerUtils.CreateLabel(labelName, indentLevel));
+            propertyTextField = new TextField(512, false, false, ' ') { isDelayed = true };
+            propertyRow.Add((TextField)propertyTextField,
+            textField =>
+            {
+                textField.value = fieldToDraw;
+            });
+
+            if (valueChangedCallback != null)
+            {
+                var textField = (TextField) propertyTextField;
+                textField.RegisterValueChangedCallback(evt => valueChangedCallback(evt.newValue));
+            }
+
+            propertyRow.styleSheets.Add(Resources.Load<StyleSheet>("Styles/PropertyRow"));
+            return propertyRow;
+        }
+
+        public Action inspectorUpdateDelegate { get; set; }
+
+        public VisualElement DrawProperty(PropertyInfo propertyInfo, object actualObject, InspectableAttribute attribute)
+        {
+            return this.CreateGUI(
+                // Use the setter from the provided property as the callback
+                newStringValue => propertyInfo.GetSetMethod(true).Invoke(actualObject, new object[] {newStringValue}),
+                (string) propertyInfo.GetValue(actualObject),
+                attribute.labelName,
+                out var propertyVisualElement);
+        }
+    }
+
+    [SGPropertyDrawer(typeof(IEnumerable<string>))]
+    class TextArrayPropertyDrawer : IPropertyDrawer
```

Comment:
Not sure TextArray is the best name here; I think DropdownPropertyDrawer or PopupPropertyDrawer is clearer from the user's side. You could display the same list of strings in different ways and this one is specifically a dropdown.

Reply:
Replaced the name with DropdownPropertyDrawer